tests/compat.rs

```diff
@@ -213,66 +213,12 @@ rpm -vv --checksig /out/{rpm_file} 2>&1
 
         podman_container_launcher(cmd.as_str(), "fedora:38", vec![])?;
 
-        let out_file = std::fs::File::open(&out_file).expect("should be able to open rpm file");
-        let mut buf_reader = std::io::BufReader::new(out_file);
-        let package = rpm::Package::parse(&mut buf_reader)?;
+        let package = rpm::Package::open(&out_file)?;
 
         package.verify_signature(verifier)?;
 
         Ok(())
     }
-
-    #[test]
-    #[serial_test::serial]
-    fn test_verify_raw_gpg_signature() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I didn't feel like this test was adding sufficient value

Reply:
It does verify the signature creation done with `rpmbuild` matches that of `rpm-rs`, which I do find a worthwhile thing to check against.

Could you please re-add it?

Reply:
That's what the previous test does.  What this test seems to be doing, is directly sign a bit of data with `gpg`, and verify it with `Verifier`.  RPMs aren't involved.

It's not a *bad* thing to test exactly, but it's not providing any additional coverage beyond what the other tests provide, it takes a long time to run, and it's not a simple test.  If we were to keep it I'd rather just commit a signature of one of the existing fixture files as a fixture.

Reply:
It doesn't sign an _rpm_, but it does call the signature process, and hence verifies the signing command of `rpmbuild` being consistent with the signing of `rpm-rs`.

It adds value, such that it tests exactly one thing, rather than the entire process. To improve the runtime, we should always compile the cryptographic dependency libraries with optimizations enabled.

Reply:
>It doesn't sign an rpm, but it does call the signature process, and hence verifies the signing command of rpmbuild being consistent with the signing of rpm-rs.

At no point in this test is the signing of `rpm-rs` being tested or compared against `rpmbuild`.  `Verifier` is exercised, but `Signer` is not.  One single signature is created using a shell script using GPG which we provide, which isn't necessarily consistent with either `rpm-rs` or `rpm`, and then verified.  

To do what you're discussing, you'd want to sign both ways and then directly compare the two signatures, but that isn't happening here.